src/System.Management.Automation/engine/GetCommandCommand.cs

```diff
@@ -733,7 +752,32 @@ private void AccumulateMatchingCommands(IEnumerable<string> commandNames)
                         {
                             if (TotalCount < 0 || count < TotalCount)
                             {
-                                foreach (CommandInfo command in System.Management.Automation.Internal.ModuleUtils.GetMatchingCommands(plainCommandName, this.Context, this.MyInvocation.CommandOrigin, rediscoverImportedModules: true, moduleVersionRequired: _isFullyQualifiedModuleSpecified))
+                                IEnumerable<CommandInfo> commands;
+                                if (UseFuzzyMatching)
+                                {
+                                    foreach (Tuple<CommandInfo, int> commandScore in System.Management.Automation.Internal.ModuleUtils.GetFuzzyMatchingCommands(
```

Comment:
Given that the lexical stack is now `method` > `foreach` > `try` > `if` > `else if` > `if` > `if`, I think we should seriously consider spinning out at least one method here. One that returns an `IEnumerable<CommandInfo>` maybe?